docs/modules/ROOT/pages/user-code-namespaces.adoc

```diff
@@ -0,0 +1,35 @@
+= User Code Namespaces
+:description: User Code Namespaces provide a container for Java classpath resources, such as user code and accompanying artifacts like property files. This provides namespace isolation to ensure that access to resources in different namespaces can be managed through configuration. From Hazelcast Platform Operator, you can deploy your application code from external buckets using UserCodeNamespace CRs. 
+
+{description}
+
+See xref:hazelcast:clusters:user-code-namespaces.adoc[User Code Namespaces] for more detailed information about the configuration and behavior of User Code Namespaces in Hazelcast.
+
+== Enabling User Code Namespaces
+
+To enable User Code Namespaces you must configure the `userCodeNamespaces` section in the Hazelcast custom resource.
+
+.Example of Hazelcast with User Code Namespaces enabled
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/hazelcast-ucn.yaml[]
+----
+
+== Creating a new User Code Namespace
+
+To create a new User Code Namespace you must create a new `UserCodeNamespace` CR. The name of the CR will be used as the name of the created User Code Namespace. The JAR files stored in the configured bucket will be uploaded to the Hazelcast cluster under the newly created namespace.
+
+.Example of User Code Namespaces
```

Comment:
Shall we add multiple tabs for different cloud providers (AWS, GCP, Azure) as we do in backup and restore docs?